nisystemlink/clients/spec/utilities/_dataframe_utility.py

```diff
@@ -0,0 +1,266 @@
+    specs = []
+
+    if spec_response.specs:
+        specs = spec_response.specs
+
+    while spec_response.continuation_token:
+        query_request.continuation_token = spec_response.continuation_token
+        spec_response = client.query_specs(query_request)
+
+        if spec_response.specs:
+            specs.extend(spec_response.specs)
+
+    return specs
+
+
+def get_specs_dataframe(
+    client: SpecClient,
+    product_id: str,
+    filter: Optional[str] = None,
+    column_projection: Optional[List[SpecificationProjection]] = None,
+    condition_format: Callable[
+        [List[Condition]], Dict[str, str]
+    ] = __serialize_conditions,
+) -> pd.DataFrame:
+    """Get specs of a specific product as a dataframe.
+
+    Args:
+        client: The Spec Client to use for the request.
+        product_ids: ID of the product to query specs.
+        filter: The specification query filter in Dynamic Linq format.
+        column_projection: List of columns to be included to the spec dataframe.
+                           This is an optional parameter. By default all the values will be retrieved.
+        condition_format: Function which takes in a list of condition objects and returns
+                          a dictionary of condition and its values. The dictionary keys
+                          should be the condition name and the values should be the condition
+                          value in any format you need. Keys will be used as the dataframe
+                          column header and values will be used as the row cells for the
+                          respective column header.
+                          This is an optional parameter. By default column header will be
+                          "condition_conditionName(conditionUnit)".
+                          The column value will be "[min: num; max: num, step: num],
+                          [min: num; max: num, step: num], num, num" where data within the '[]' is
+                          numeric condition range and other num values are numeric condition discrete values.
+                          The column value will be "str, str, str" - where str values are the
+                          condition discrete values for a string condition.
+
+    Returns:
+        The list of specs of the specified product as a dataframe.
+
+    Raises:
+        ApiException: if unable to communicate with the `/nispec` service or if there are
+        invalid arguments.
+    """
+    specs = __batch_query_specs(
+        client=client,
+        product_id=product_id,
+        filter=filter,
+        column_projection=column_projection,
+    )
+    if column_projection and (
+        SpecificationProjection.CONDITION_NAME not in column_projection
+        or SpecificationProjection.CONDITION_VALUES not in column_projection
+    ):
+        specs_df = __serialize_specs(specs=specs)
+    else:
+        specs_df = __serialize_specs(specs=specs, condition_format=condition_format)
```

Comment:
```suggestion
        specs_df = __serialize_specs(
        specs=specs, 
        condition_format=None if should_not_format_condition else condition_format
    )
```

set the logic for should_not_format_condition